cmd/gossamer/main.go

```diff
@@ -128,18 +130,61 @@ func gossamer(ctx *cli.Context) error {
 		return err
 	}
 
+	// check command arguments
 	if arguments := ctx.Args(); len(arguments) > 0 {
 		return fmt.Errorf("failed to read command argument: %q", arguments[0])
 	}
 
+	// check if node has been initialized
+	if !nodeInitialized(ctx) {
+
+		// initialize node (load genesis data into database)
+		err = initializeNode(ctx)
+		if err != nil {
+			log.Error("[gossamer] Failed to initialize node", "error", err)
+			return err
+		}
+	}
+
+	// create node services
 	node, _, err := makeNode(ctx)
 	if err != nil {
-		log.Error("[gossamer] Failed to initialize node", "error", err)
+		log.Error("[gossamer] Failed to create node services", "error", err)
 		return err
 	}
 
-	log.Info("[gossamer] Starting node services...", "name", node.Name)
+	log.Info("[gossamer] Starting node...", "name", node.Name)
+
+	// start node
 	node.Start()
 
 	return nil
 }
+
+// nodeInitialized returns true if, within the configured data directory for the
+// node, the state database has been created and the gensis data has been loaded
+func nodeInitialized(ctx *cli.Context) bool {
```

Comment:
is there a better way to check if a node is initialized without being too costly (ie, spinning up the state service)?